### ISSUE TITLE
Non latin filename triggers codecs error with CherryPy 5.5.0 or later

### ISSUE THREAD

**Issue by atitan** [devsnd/cherrymusic#632]
Hi there,

I have problem playing songs with CJK characters in their filename. Removing the URI workaround or renaming the files to English will work.

After some research, I found the cause was CherryPy. The URI workaround is needed before 5.5.0, however, seems they have fixed the issue and the workaround will cause error since version 5.5.0.

Environment:
Fedora 24 with kernel 4.5.7
Python 3.5.1
Nginx 1.10.1 (reverse proxy)
Cherrymusic 6841e74 from devel branch

```
[160704-19:57] 127.0.0.1 - - [04/Jul/2016:19:57:18] "GET /serve/1.\xe4\xba\x88\xe6\x84\x9f.mp3 HTTP/1.0" 500 825
[160704-19:57] ERROR   : [04/Jul/2016:19:57:18]
Traceback (most recent call last):
 File "/home/cherrymusic/cherrymusic/cherrypy/_cprequest.py", line 104, in run
   hook()
 File "/home/cherrymusic/cherrymusic/cherrypy/_cprequest.py", line 63, in __call__
   return self.callback(**self.kwargs)
 File "/home/cherrymusic/cherrymusic/cherrypy/_cptools.py", line 184, in _wrapper
   if self.callable(**kwargs):
 File "/home/cherrymusic/cherrymusic/cherrypy/lib/static.py", line 335, in staticdir
   handled = _attempt(filename, content_types)
 File "/home/cherrymusic/cherrymusic/cherrypy/lib/static.py", line 261, in _attempt
   serve_file(filename, content_type=content_type, debug=debug)
 File "/home/cherrymusic/cherrymusic/cherrymusicserver/__init__.py", line 160, in serve_file_utf8_fix
   path = codecs.decode(codecs.encode(path, 'latin-1'), 'utf-8')
UnicodeEncodeError: 'latin-1' codec can't encode characters in position 26-27: ordinal not in range(256)
```

**Comment by Isgar** [comment]
I can confirm, downgrading to python-cherrypy 5.4.0 fixed the issue for me (Arch Linux)

**Comment by devsnd** [comment]
Hey Sh Lin, hi Isgar,

Thanks so much for you bug report and all the details contained! Using your information I could pinpoint the issue easily. I have written a patch that is now in devel, that will apply the cherrypy monkey-patch only for versions <= 5.4.

Can you please try out the devel branch and report back if it worked for you as well? If so, I can release a new patch-version.

**Comment by atitan** [comment]
It seems the variable `needs_serve_file_utf8_fix` is not correctly assigned. The problem still exists.

**Comment by devsnd** [comment]
@atitan Can you please paste the output of `cherrymusic --info`? I wonder why the detection of your cherrypy version did not work....

**Comment by atitan** [comment]
My CherryPy was from pypi website, extracted and moved `cherrypy` folder into cherrymusic manually.

After I fixed pip3(it was broken before) and installed CherryPy using pip3, now the version is correct.

```
Could not determine cherrypy version. Please install cherrypy using pip or your OS's package manager. Trying to detect version automatically.
CherryMusic Server 0.37.0

CherryPy: unknown
Python: CPython 3.5.1
Platform: Linux-4.6.3-300.fc24.x86_64-x86_64-with-fedora-24-Twenty_Four

configuration dir:
    /home/cherrymusic/.config/cherrymusic
server data dir:
    /home/cherrymusic/.local/share/cherrymusic
static resources dir:
    /home/cherrymusic/cherrymusic/
server package dir:
    /home/cherrymusic/cherrymusic/cherrymusicserver
process working dir:
    /home/cherrymusic

locale: ('en_US', 'UTF-8'), default: ('en_US', 'UTF-8')
filesystem encoding: utf-8

Available Decoders:
    wav (cat)
Available Encoders:
    wav (cat)
```

**Comment by devsnd** [comment]
Can you please report if the problem still persists in the newest version? (0.37.1)

**Comment by Isgar** [comment]
#633 is fixed, but this problem still persists.

```
>>> import cherrypy
>>> import cherrypy._cptools.register
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named 'cherrypy._cptools.register'; 'cherrypy._cptools' is not a package
```

arch linux package: python-cherrypy 6.0.2-2

**Comment by atitan** [comment]
The latest commit acts the same as I've mentioned before.

CherryPy version tested: 7.1.0

Install CherryPy using pip: works
Install CherryPy by extracting directory into CherryMusic: unknown version

**Comment by Isgar** [comment]
sorry, the fix was incomplete, i updated the pull request

somehow got mixed up while testing

**Comment by devsnd** [comment]
Hey @Isgar,

Thanks so much for the pull-request. I just merged it into devel and will release a new version soon.

All this trouble just because cherrypy is too lazy to keep the version updated inside the sources... :disappointed:

**Comment by devsnd** [comment]
I released a version 0.37.2 that contains your fix, thanks @Isgar!
